src/pygeosx/pyhdf5/PyHDF5.cpp

```diff
@@ -0,0 +1,294 @@
+
+  if( !PyArg_ParseTuple( args, "Udd", &unicodePath, &time, &dt ) )
+  {
+    return nullptr;
+  }
+
+  LvArray::python::PyObjectRef<> asciiPath { PyUnicode_AsASCIIString( unicodePath ) };
+  if( asciiPath == nullptr )
+  {
+    return nullptr;
+  }
+
+  char const * const key = PyBytes_AsString( asciiPath );
+  if( key == nullptr )
+  {
+    return nullptr;
+  }
+
+  geosx::DomainPartition & domain = self->pb_manager->getDomainPartition();
+
+  try
+  {
+    self->collection.at( key )->geosx::HistoryCollection::execute( time, dt, 0, 0, 0, domain );
+  }
+  catch( std::out_of_range const & e )
+  {
+    std::cout<<"Target \""<<key<<"\" not found. Impossible output."<<std::endl;
+  }
+  Py_RETURN_NONE;
+}
+
+
+static PyObject * output( PyHDF5 * self, PyObject * args )
+{
+  VERIFY_NON_NULL_SELF( self );
+  VERIFY_INITIALIZED( self );
+
+  double time;
+  double dt;
+  PyObject * unicodePath;
+
+  if( !PyArg_ParseTuple( args, "Udd", &unicodePath, &time, &dt ) )
+  {
+    return nullptr;
+  }
+
+  LvArray::python::PyObjectRef<> asciiPath { PyUnicode_AsASCIIString( unicodePath ) };
+  if( asciiPath == nullptr )
+  {
+    return nullptr;
+  }
+
+  char const * const key = PyBytes_AsString( asciiPath );
+  if( key == nullptr )
+  {
+    return nullptr;
+  }
+
+  geosx::DomainPartition & domain = self->pb_manager->getDomainPartition();
+
+  try
+  {
+    self->output.at( key )->geosx::TimeHistoryOutput::execute( time, dt, 0, 0, 0, domain );
+  }
+  catch( std::out_of_range const & e )
+  {
+    std::cout<<"Target \""<<key<<"\" not found. Impossible output."<<std::endl;
+  }
+  Py_RETURN_NONE;
+}
+
+static PyMethodDef PyHDF5_methods[] = {
+  { "collect", (PyCFunction) collect, METH_VARARGS, "wrapper to routine TimeHistoryCollection::execute" },
+  { "output", (PyCFunction) output, METH_VARARGS, "wrapper to routine TimeHistoryOutput::execute"},
+  { nullptr, nullptr, 0, nullptr }      /* Sentinel */
+};
+
+
+/**
+ * Initialize the module object for Python with the exported functions
+ */
+
+BEGIN_ALLOW_DESIGNATED_INITIALIZERS
+
+static PyTypeObject PyHDF5Type = {
+  PyVarObject_HEAD_INIT( nullptr, 0 )
+    .tp_name = "pysolver.HDF5",
+  .tp_basicsize = sizeof( PyHDF5 ),
+  .tp_itemsize = 0,
+  .tp_dealloc = (destructor)PyHDF5_dealloc,
+  .tp_repr = PyHDF5_repr,
+  .tp_flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
+  .tp_doc = PyHDF5::docString,
+  .tp_methods = PyHDF5_methods,
+  .tp_init = (initproc)PyHDF5_init,
+  .tp_new = PyHDF5_new,
+};
+
+END_ALLOW_DESIGNATED_INITIALIZERS
+
+static PyModuleDef pyhdf5module = {
+  PyModuleDef_HEAD_INIT,
+  "pyhdf5",
+  "pyhdf5 module for HDF5",
+  -1,
+  NULL, NULL, NULL, NULL, NULL
+};
+
+
+PyMODINIT_FUNC
+PyInit_pyhdf5( void )
+{
+  PyObject * module = PyModule_Create( &pyhdf5module );
+
+  if( PyType_Ready( &PyHDF5Type ) < 0 )
+    return nullptr;
+
+  if( module == nullptr )
+    return nullptr;
+
+  Py_INCREF( &PyHDF5Type );
+  PyModule_AddObject( module, "HDF5", (PyObject *)&PyHDF5Type );
```

Comment:
I'd rewrite these using `PyObectRef` to eliminate leaks.

Reply:
Done

Reply:
Actually when I try to call `PyModule_Create()` using the `PyObjectRef` for `PySolver` and `PyHistory` in `PySolver.cpp` and `PyHistory.cpp` (ex :   LvArray::python::PyObjectRef<> module{ PyModule_Create( &pysolvermodule ) }; ) 
This is causing a `core dumped` at execution.

Reply:
You have to be careful with how you do it, if I had to guess you're missing a call to `release` and the module is getting deleted when it shouldn't be. Have you pushed the breaking change?

Reply:
Yes I pushed it

Reply:
I think I should not `Py_XINCREF` in `pygeosx.cpp` since you're doing it `PyObjectRef` ?

Reply:
I don't see it in the changes, I don't even see `PyHDF5.cpp` or `HistoryCollection.cpp`.

Reply:
It's strange because I can see it on GitHub. I will push something now.

Reply:
I pushed something with the breaking change and the "fix" line commented. I also pushed a dummy example with the xml associated with it. It's all in the src/pygeosx directory.